### PR TITLE
Include git commit hash in docker image tags

### PR DESCRIPTION
**Pull request by jodersky**

### CODE REVIEW

File: src/main/scala/xyz.driver.sbt/Service.scala [hunk 1]
```diff
@@ -64,7 +58,7 @@ object Service extends AutoPlugin {
       crossScalaVersions := List("2.12.6"),
       scalaVersion := crossScalaVersions.value.last,
       publish := {
-        streams.value.log.warn(s"Service ${name.value} won't be published.")
+        streams.value.log.warn(s"Project ${name.value} is a service and will not be published to an artifactory.")
```
Comment (jodersky) [inline src/main/scala/xyz.driver.sbt/Service.scala]: unrelated, but I think the previous error message was somewhat confusing

File: src/main/scala/xyz.driver.sbt/Service.scala [hunk 2]
```diff
@@ -64,7 +58,8 @@ object Service extends AutoPlugin {
       crossScalaVersions := List("2.12.6"),
       scalaVersion := crossScalaVersions.value.last,
       publish := {
-        streams.value.log.warn(s"Service ${name.value} won't be published.")
+        streams.value.log
+          .warn(s"Project ${name.value} is a service and will therefore not be published to an artifactory.")
```
Comment (jodersky) [inline src/main/scala/xyz.driver.sbt/Service.scala:65]: unrelated, but I think the previous error message was somewhat confusing

File: src/main/scala/xyz.driver.sbt/Service.scala [hunk 3]
```diff
@@ -41,6 +34,7 @@ object Service extends AutoPlugin {
     dockerRepository := Some("gcr.io/driverinc-sandbox"),
     dockerUpdateLatest := true, // automatically update the latest tag
     dockerBaseImage := "openjdk:10",
+    dockerLabels := Map("commit" -> git.gitHeadCommit.value.getOrElse("0000000000000000000000000000000000000000")),
```
Comment (sboora) [inline src/main/scala/xyz.driver.sbt/Service.scala]: nit: may be we can use some sort of date version as a fallback instead of all `0`s

**Review by sboora — APPROVED** [review]

File: src/main/scala/xyz.driver.sbt/Service.scala (hunk 3 above)
Comment (jodersky) [reply src/main/scala/xyz.driver.sbt/Service.scala]: Good idea to add a build date too! I'd do that in another field though.
What if we ommit the field entirely if there is no commit available?

**Review by sboora — APPROVED** [review]

File: src/main/scala/xyz.driver.sbt/Service.scala (hunk 2 above)
Comment (zachdriver) [reply src/main/scala/xyz.driver.sbt/Service.scala:65]: Out of curiosity, would it be possible to reuse `sbt publish` to push docker images to GCR (basically replacing `dr docker push`)? Could be kind of cool...

**Pull request merged**